src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -862,15 +862,22 @@ private static JsonRpcMessage CreateError(JToken id, Exception exception)
                 throw new ArgumentNullException(nameof(exception));
             }
 
+            exception = StripExceptionToInnerException(exception);
+
+            var data = new { stack = exception.StackTrace, code = exception.HResult.ToString(CultureInfo.InvariantCulture) };
+            return JsonRpcMessage.CreateError(id, JsonRpcErrorCode.InvocationError, exception.Message, JObject.FromObject(data, DefaultJsonSerializer));
+        }
+
+        private static Exception StripExceptionToInnerException(Exception exception)
+        {
             if (exception is TargetInvocationException || (exception is AggregateException && exception.InnerException != null))
```

Comment:
This helper method looks much better. I'm a little concerned about inconsistent handling of AggregateException, which we can talk about in person since I find it hard to explain without illustrations.

Maybe I'll just push a couple new test methods to your branch to demonstrate what I believe may be a problem.